djangocms_picture/cms_plugins.py

```diff
@@ -1,46 +1,77 @@
 from django.utils.translation import ugettext_lazy as _
 
 from cms.plugin_base import CMSPluginBase
 from cms.plugin_pool import plugin_pool
 
 from .models import Picture
+from .forms import PictureForm
+
+
+# enable nesting of plugins inside the picture plugin
+PICTURE_NESTING = getattr(settings, 'DJANGOCMS_PICTURE_NESTING', False)
 
 
 class PicturePlugin(CMSPluginBase):
     model = Picture
-    name = _("Picture")
-    render_template = "cms/plugins/picture.html"
+    form = PictureForm
+    name = _('Image')
+    allow_children = PICTURE_NESTING
     text_enabled = True
 
-    def render(self, context, instance, placeholder):
-        if instance.url:
-            link = instance.url
-        elif instance.page_link:
-            link = instance.page_link.get_absolute_url()
-        else:
-            link = ""
-        context.update({
-            'picture': instance,
-            'link': link,
-            'placeholder': placeholder
+    fieldsets = [
+        (None, {
+            'fields': (
+                'picture',
+                'external_picture',
+            )
+        }),
+        (_('Advanced settings'), {
+            'classes': ('collapse',),
+            'fields': (
+                'template',
+                ('width', 'height'),
+                'alignment',
+                'caption_text',
+                'attributes',
+            )
+        }),
+        (_('Link settings'), {
+            'classes': ('collapse',),
+            'fields': (
+                ('link_url', 'link_page'),
+                'link_target',
+                'link_attributes',
+            )
+        }),
+        (_('Cropping settings'), {
+            'classes': ('collapse',),
+            'fields': (
+                ('use_automatic_scaling', 'use_no_cropping'),
+                ('use_crop', 'use_upscale'),
+                'use_thumbnail',
+            )
         })
-        return context
-
-    def icon_src(self, instance):
-        if getattr(settings, 'PICTURE_FULL_IMAGE_AS_ICON', False):
-            return instance.image.url
-        else:
-            return urlparse.urljoin(
-                settings.STATIC_URL, "cms/img/icons/plugins/image.png")
+    ]
+
+    def get_render_template(self, context, instance, placeholder):
+        return 'djangocms_picture/{}/picture.html'.format(instance.template)
+
+    def render(self, context, instance, placeholder):
+        classes = ''
+        if instance.alignment:
+            classes += 'align-{} '.format(instance.alignment)
+        if 'class' in instance.attributes:
+            classes += '{} '.format(instance.attributes['class'])
+        # we actually want to modify the class in attributes
+        # this saves a simple if/else in the template
+        instance.attributes['class'] = classes
+        # assign link to a context variable to be performant
+        context['get_link'] = instance.get_link
+        context['get_size'] = instance.get_size(context, placeholder)
```

Comment:
shouldn't it be size instead of `get_size`? `get_` is reserved for getter methods